File: c++/Hardware.cpp
#include <bits/stdc++.h>
#include "Product.cpp"

using namespace std;

class Hardware : public Product
{
private:
    string brand;
    string model;

public:
    Hardware()
    {
        this->brand = "-";
        this->model = "-";
    }

    Hardware(string brand, string model)
    {
        this->brand = brand;
        this->model = model;
    }

    void setBrand(string brand)
    {
        this->brand = brand;
    }

    void setModel(string model)
    {
        this->model = model;
    }

    string getBrand()
    {
        return this->brand;
    }

    string getModel()
    {
        return this->model;
    }

    ~Hardware(){};
};

File: PHP/Hardware.php
<?php

include "Product.php";

class Hardware extends Product
{
    private $brand, $model;

    public function __construct($brand = "-", $model = "-")
    {
        $this->brand = $brand;
        $this->model = $model;
    }

    public function setBrand($brand)
    {
        $this->brand = $brand;
    }

    public function setModel($model)
    {
        $this->model = $model;
    }

    public function getBrand()
    {
        return $this->brand;
    }

    public function getModel()
    {
        return $this->model;
    }
}